TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -723,19 +725,38 @@ private static SymbolReference ResolveProcedureName(SymbolTable table, SymbolRef
             return null;
         }
 
-        protected static void Check<T>(T node, [NotNull] IList<T> found) where T : Node
+        protected static void Check<T>(string nodeTypeName, T node, [NotNull] IList<T> found) where T : Node
         {
-            if (found.Count > 1) DiagnosticUtils.AddError(node, "Symbol \'" + node.Name + "\' already declared");
+            if (found.Count > 1)
+                DiagnosticUtils.AddError(node, nodeTypeName + " \'" + node.Name + "\' already declared");
+
+            // a section/paragraph is empty when it has no child or when its child/children is/are an {End} node
+            bool empty = true;
+            int nodesCount = node.Children.Count; 
+            // checks if all children are of type {End} : if not one then empty = false 
+            for (int i = 0; i < nodesCount; i++)
+            {
+                if  ((node.Children[i].Children.Count == 1 && node.Children[i].Children[0].GetType() == typeof(Nodes.End)) == false)
```

Comment:
Can you add a comment to explain why you test `node.Children[i].Children.Count == 1`? I guess it's because of Sentence nodes.

Reply:
Ok, but what is the type of "node.Children[i]" ?
One should think that an empty paragraph contains no Children or maybe just a child of type End.
So the test would be:
`if  ((node.Children.Count == 1 && node.Children[0].GetType() == typeof(Nodes.End)) == false)`

Reply:
done by using explicit word "sentence"
https://github.com/TypeCobolTeam/TypeCobol/pull/1479/commits/90e9535abb7b0fc6faada8125915a1be4e97a24a